src/components/Tooltip/index.js

```diff
@@ -91,46 +54,30 @@ class Tooltip extends PureComponent {
         if (!this.state.isRendered) {
             this.setState({isRendered: true});
         }
+
         this.animation.stopAnimation();
-        this.shouldStartShowAnimation = true;
-
-        // We have to dynamically calculate the position here as tooltip could have been rendered on some elments
-        // that has changed its position
-        this.getWrapperPositionPromise = makeCancellablePromise(this.getWrapperPosition());
-        this.getWrapperPositionPromise.promise.then(({x, y, width, height}) => {
-            this.setState({
-                wrapperWidth: width,
-                wrapperHeight: height,
-                xOffset: x,
-                yOffset: y,
-            });
 
-            // We may need this check due to the reason that the animation start will fire async
-            // and hideTooltip could fire before it thus keeping the Tooltip visible
-            if (this.shouldStartShowAnimation) {
```

Comment:
@parasharrajat Please note that this check nearly is directly below the `this.shouldStartShowAnimation = true` assignment. The only barrier between is the awaiting for `this.getWrapperPositionPromise.promise` it's gone, so the `shouldStartShowAnimation` flag doesn't have any use anymore, as I see it.